cirq-superstaq/cirq_superstaq/ops/qubit_gates.py

```diff
@@ -979,6 +979,66 @@ def _json_dict_(self) -> dict[str, Any]:
         return cirq.obj_to_dict_helper(self, ["rz_rads"])
 
 
+class DDPowGate(cirq.EigenGate):
+    r"""The Dipole-Dipole gate for EeroQ hardware"""
+
+    def _eigen_components(self) -> list[tuple[float, npt.NDArray[np.float_]]]:
+        return [
+            (
+                -0.5,
+                np.array([[1, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 1]]),
+            ),
+            (
+                1.0,
+                np.array([[0, 0, 0, 0], [0, 0.5, 0.5, 0], [0, 0.5, 0.5, 0], [0, 0, 0, 0]]),
+            ),
+            (
+                0.0,
+                np.array([[0, 0, 0, 0], [0, 0.5, -0.5, 0], [0, -0.5, 0.5, 0], [0, 0, 0, 0]]),
+            ),
+        ]
+
+    def _num_qubits_(self) -> int:
+        return 2
+
+    def _decompose_(self, qubits: tuple[cirq.LineQubit, cirq.LineQubit]) -> list[cirq.Operation]:
+        return [
+            cirq.ZZPowGate(exponent=self.exponent, global_shift=self.global_shift - 0.5).on(
+                *qubits
+            ),
+            cirq.ISwapPowGate(exponent=self.exponent).on(*qubits),
+        ]
+
+    def _has_stabilizer_effect_(self) -> bool | None:
+        if cirq.is_parameterized(self):
+            return None
+
+        return (2 * self.exponent) % 1 == 0
+
+    def _circuit_diagram_info_(self, args: cirq.CircuitDiagramInfoArgs) -> cirq.CircuitDiagramInfo:
+        return cirq.CircuitDiagramInfo(
+            wire_symbols=("DD", "DD"), exponent=self._diagram_exponent(args)
+        )
+
+    def __str__(self) -> str:
+        if self.exponent == 1:
+            return "DD"
+        return f"DD**{self._exponent!r}"
```

Comment:
nit: parenthesis, and no need for the "!r" in `__str__`
```suggestion
        return f"(DD**{self._exponent})"
```